src/jit/namedintrinsiclist.h

```diff
@@ -16,10 +16,70 @@ enum NamedIntrinsic : unsigned int
     NI_System_Collections_Generic_EqualityComparer_get_Default = 4,
 #if FEATURE_HW_INTRINSICS
     NI_HW_INTRINSIC_START,
-#define HARDWARE_INTRINSIC(id, name, isa) NI_##id,
+#define HARDWARE_INTRINSIC(id, name, isa, ret, num, ival, size, t1, t2, t3, t4, t5, t6, t7, t8, t9, t10, cate) NI_##id,
 #include "hwintrinsiclistxarch.h"
     NI_HW_INTRINSIC_END
 #endif
 };
 
+#if FEATURE_HW_INTRINSICS && defined(_TARGET_XARCH_)
+enum HWIntrinsicCategory : unsigned int
+{
+    // Simple SIMD intrinsics
+    // - take Vector128/256<T> parameters
+    // - return a value of Vector128/256<T>
+    // - all the vector parameters and return value has same vector size
+    // - generate single instruction
+    // - the codegen of overloads can be determined by intrinsicID and base type of returned vector
+    // - not any other kind intrinsics
+    HW_Category_SimpleSIMD = 0,
+
+    // Scalar intrinsics
+    HW_Category_Scalar = 0x1,
```

Comment:
What does "scalar" imply here? We have scalar intrinsics (like `SqrtScalar`), but I think these are meant to cover the intrinsics like `popcnt`, `lzcnt`, and `crc32`.

Reply:
Yes.

Reply:
An explicit comment might help, but I think we should come up with a different name to distinguish it from the other scalar intrinsics.

Reply:
Good point, will change the names.